src/Microsoft.DotNet.Wpf/src/PresentationBuildTasks/MS/Internal/Tasks/IncrementalCompileAnalyzer.cs

```diff
@@ -389,13 +389,22 @@ private void UpdateFileListForIncrementalBuild(List<FileUnit> modifiedXamlFiles)
                 {
                     numLocalTypeXamls = CompilerLocalReference.LocalMarkupPages.Length;
 
+                    // Under incremental builds of SDK projects, we can have a state where the cache contains some XAML files but the Page blob
+                    // no longer contains them.  To avoid attempting to recompile a file that no longer exists, ensure that any cached XAML file
+                    // still exists in the Page blob prior to queuing it up for recompilation.
+                    HashSet<string> localMarkupPages = new HashSet<string>(_mcPass1.PageMarkup.Select(x => x.GetMetadata("FullPath")));
```

Comment:
Instantiate `HashSet<string>` with `StringComparer.OrdinalIgnoreCase`, something like this:
```C#
HashSet<string> localMarkupPages = new HashSet<string>(_mcPass1.PageMarkup.Select(x => x.GetMetadata("FullPath")), StringComparer.OrdinalIgnoreCase);

```